File: 010/euler-010.c
/*
Find the sum of all primes below two million.

Uses slightly optimized trial division. Compile as C99 to avoid any
warnings from the use of uint64.
*/

#include <stdio.h>
#include <stdint.h>
#include <inttypes.h>
#include <stdbool.h>

bool is_prime(unsigned n)
{
	uint64_t i;

	if (n == 2 || (n % 2) == 0 || n < 2)
		return false;

	for (i = 3; i <= n/i; i += 2) {
		if ((n % i) == 0)
			return false;
	}

	return true;
}

int main(void)
{
	uint64_t sum = 0;
	uint64_t n;
	
	for (n = 2; n < 2000000; n++) {
		if (is_prime(n)) {
			sum += n;
		}
	}

	printf("%" PRIu64 "\n", sum);	
	return 0;
}

File: 010/euler-010.py
# Find the sum of all primes below two million.
# Equivalent to the C version, but much slower.

def is_prime(n):
	if n == 2 or n % 2 == 0 or n < 2:
		return False

	for i in range(3, n ** 1/2 + 1, 2):
		if n % i == 0:
			return False

	return True

sum = 0
n = 2
while n < 2000000:
	if is_prime(n):
		sum += n
	n += 1

print sum
